### PR TITLE
test(browse-ci): add unit tests

### LINKED ISSUE #1128: Test(git-browse-ci): add unit tests
## File
[git-browse-ci](https://github.com/tj/git-extras/blob/main/bin/git-browse-ci)

`git browse-ci` is that the command browses the repository of the CI page which comes from gitlab, github or bitbucket.

## Testcases
1. If the repository comes from gitlab, open a url like `https://gitlab.com/<user_or_group>/<repo>/-/pipelines`
2. If the repository comes from github, open a url like `https://github.com/<user_or_org>/<repo>/actions`
3. If the repository comes from bitbucket, open a url like `https://bitbucket.org/<user_or_org>/<repo>/addon/pipelines/home`
4. If the OS is mac, open via command `open`
5. If the OS is Git-Bash on windows, open via command `start`
6. If the OS is WSL on windows, open via commnd `start`
7. If the OS is linux, open via command `xdg-open`
8. If the OS is unknown, open via command `xdg-open`

PS: it is simalir with `git-browse` but for different purposes.

### PR DESCRIPTION
**Pull request by vanpipy**
Close #1128

### CODE REVIEW

**Review by spacewander — APPROVED** [review]

**Comment by spacewander** [comment]
Sorry for not handling it on time! I am too busy this week, just get some sparse time at the weekend.

**Comment by spacewander** [comment]
After this PR is merged, @vanpipy you can refactor it via some metaprogramming (maybe using a function builder or something else). This will make the code simpler.

File: tests/test_git_browse_ci.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+        return github_origin + '/actions'
+    if mode == "gitlab":
+        return gitlab_origin + '/-/pipelines'
+    if mode == "bitbucket":
+        return bitbucket_origin + '/addon/pipelines/home'
+
+class TestGitBrowse:
+    def test_browse_github_ci_on_mac(self, temp_repo):
+        git = temp_repo.get_repo_git()
+        git.remote("add", "origin", github_origin + ".git")
```
Comment (hyperupcall) [inline tests/test_git_browse_ci.py:23]: I'm having trouble figuring out why some tests use `set_origin_url` and others don't. I thought a new Git repository was created for each test, so "git remote add" should be done for all of them?

Comment (vanpipy) [reply tests/test_git_browse_ci.py:23]: It should be all used with `set_origin_url` or a code error.

Comment (vanpipy) [reply tests/test_git_browse_ci.py:23]: I got your meaning, emmm, 

> I thought a new Git repository was created for each test
Nope, a test repository for testing one command

> so "git remote add" should be done for all of them?
One temp repository created for testing one command , so initialize it once and then use it again.

Comment (vanpipy) [reply tests/test_git_browse_ci.py:23]: It is my bad without any document to describe it, I will fill it later with some refactors.

**Review by hyperupcall — APPROVED** [review]

**Pull request merged**